repos/spack_repo/builtin/packages/tcl_togl/package.py

```diff
@@ -0,0 +1,67 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack_repo.builtin.build_systems.autotools import AutotoolsPackage
+
+from spack.package import *
+
+
+class TclTogl(AutotoolsPackage):
+    """Tcl-Togl provides a platform independent Tcl/Tk widget
+    for using OpenGL rendering contexts"""
+
+    homepage = "https://togl.sourceforge.net/"
+    url = "https://downloads.sourceforge.net/project/togl/Togl/2.0/Togl2.0-src.tar.gz"
+
+    version("2.0", sha256="b7d4a90bbad3aca618d505ee99e7fd8fb04c829f63231dda2360f557ba3f7610")
+
+    variant("64bit", default=False, description="Build and link with shared libraries")
+    variant("64bit-vis", default=False, description="Enable 64bit Sparc VIS support")
+    variant("load", default=True, description='Allow dynamic loading and "load" command')
+    variant("rpath", default=False, description="Disable rpath support")
+    variant("shared", default=True, description="Build and link with shared libraries")
+    variant("stubs", default=True, description="Build and link with stub libraries")
+    variant("symbols", default=True, description="Build with debugging symbols")
+    variant("threads", default=True, description="Build with threads")
+    variant("wince", default=False, description="Enable Win/CE support")
+
+    depends_on("autoconf", type="build", when="build_system=autotools")
+    depends_on("automake", type="build", when="build_system=autotools")
+    depends_on("libtool", type="build", when="build_system=autotools")
+    depends_on("mesa~llvm")
```

Comment:
FYI. You can "tighten this up" using one or two context managers (see the docs at https://spack.readthedocs.io/en/latest/packaging_guide_creation.html#grouping-directives). For example,

```suggestion
    with when("build_system=autotools"):
        depends_on("autoconf", type="build")
        depends_on("automake", type="build")
        depends_on("libtool", type="build")

    depends_on("mesa~llvm")
```

Or, using nested context managers:
```suggestion
    with when("build_system=autotools"):
        with default_args(type="build"):
            depends_on("autoconf")
            depends_on("automake")
            depends_on("libtool")

    depends_on("mesa~llvm")
```

Reply:
Added.